clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -0,0 +1,231 @@
+}
+
+func waitForRuntimeObject(c clientset.Interface, obj runtime.Object, timeout time.Duration) error {
+	runtimeObjectNamespace, err := getNamespaceFromRuntimeObject(obj)
+	if err != nil {
+		return err
+	}
+	runtimeObjectSelector, err := getSelectorFromRuntimeObject(obj)
+	if err != nil {
+		return err
+	}
+	runtimeObjectReplicas, err := getReplicasFromRuntimeObject(obj)
+	if err != nil {
+		return err
+	}
+
+	err = waitForPods(c, runtimeObjectNamespace, runtimeObjectSelector.String(), "", int(runtimeObjectReplicas), timeout)
+	if err != nil {
+		return fmt.Errorf("waiting pods error: %v", err)
+	}
+	return nil
+}
+
+func getRuntimeObjectsForKind(c clientset.Interface, kind, namespace, labelSelector, fieldSelector string) ([]runtime.Object, error) {
+	listOpts := metav1.ListOptions{
+		LabelSelector: labelSelector,
+		FieldSelector: fieldSelector,
+	}
+	runtimeObjectsList := make([]runtime.Object, 0)
+	switch kind {
+	case "ReplicationController":
+		list, err := c.CoreV1().ReplicationControllers(namespace).List(listOpts)
+		if err != nil {
+			return runtimeObjectsList, err
+		}
+		for i := range list.Items {
+			runtimeObjectsList = append(runtimeObjectsList, &list.Items[i])
+		}
+		return runtimeObjectsList, nil
+	case "ReplicaSet":
+		list, err := c.ExtensionsV1beta1().ReplicaSets(namespace).List(listOpts)
+		if err != nil {
+			return runtimeObjectsList, err
+		}
+		for i := range list.Items {
+			runtimeObjectsList = append(runtimeObjectsList, &list.Items[i])
+		}
+		return runtimeObjectsList, nil
+	case "Deployment":
+		list, err := c.ExtensionsV1beta1().Deployments(namespace).List(listOpts)
+		if err != nil {
+			return runtimeObjectsList, err
+		}
+		for i := range list.Items {
+			runtimeObjectsList = append(runtimeObjectsList, &list.Items[i])
+		}
+		return runtimeObjectsList, nil
+	case "DaemonSet":
+		list, err := c.ExtensionsV1beta1().DaemonSets(namespace).List(listOpts)
+		if err != nil {
+			return runtimeObjectsList, err
+		}
+		for i := range list.Items {
+			runtimeObjectsList = append(runtimeObjectsList, &list.Items[i])
+		}
+		return runtimeObjectsList, nil
+	case "Job":
+		list, err := c.BatchV1().Jobs(namespace).List(listOpts)
+		if err != nil {
+			return runtimeObjectsList, err
+		}
+		for i := range list.Items {
+			runtimeObjectsList = append(runtimeObjectsList, &list.Items[i])
+		}
+		return runtimeObjectsList, nil
+	default:
+		return runtimeObjectsList, fmt.Errorf("unsupported kind when getting runtime object: %v", kind)
+	}
+}
+
+func getNamespaceFromRuntimeObject(obj runtime.Object) (string, error) {
+	metaObjectAccessor, ok := obj.(metav1.ObjectMetaAccessor)
+	if !ok {
+		return "", fmt.Errorf("unsupported kind when getting namespace: %v", obj)
+	}
+	return metaObjectAccessor.GetObjectMeta().GetNamespace(), nil
+}
+
+func getSelectorFromRuntimeObject(obj runtime.Object) (labels.Selector, error) {
```

Comment:
I think you could make this function generic by casting the obj into unstructured and accessing the selector, no?

Reply:
TBH, I would prefer not to. Current implementation uses specific structures. Which is quite convenient due to:
- verifying types of objects
- knowing the types of fields - same fields can have different types across different object kinds.

Changing this function to use unstructured objects will make it less readable.

However, if this measurement will be reimplmented to use dynamic client, using unstructured object will be preferred.

Reply:
Ok, that sounds fine. Could add some high-level TODO somewhere saying that these functions should be changed after moving to dynamic client?